### PR TITLE
generate action packages

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: cmd/cl001/ac001/explain/zz_generated.runner.go [hunk 1]
```diff
@@ -50,7 +50,12 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 		}
 	}
 
-	fmt.Println(strings.TrimSpace(e.Explain()))
+	s, err := e.Explain(ctx)
```
Comment (xh3b4sd) [inline cmd/cl001/ac001/explain/zz_generated.runner.go:53]: I extended the interface to be a bit more future prove. We can generate most of the code but it is better to not cause too many disturbances down the road. So what we can anticipate can be prevented. Having the context given and being able to return errors should make it easier to implement different kinds of explainers.

File: cmd/generate/action/runner.go [hunk 2]
```diff
@@ -45,39 +48,65 @@ func (r *runner) Run(cmd *cobra.Command, args []string) error {
...
+		// changes since go maps are not deterministic.
+		templates := [][]string{
+			{templatepkgaction.ErrorBase, templatepkgaction.ErrorContent},
+			{templatepkgaction.ExecutorBase, templatepkgaction.ExecutorContent},
+			{templatepkgaction.ExecutorCustomBase, templatepkgaction.ExecutorCustomContent},
+			{templatepkgaction.ExplainerBase, templatepkgaction.ExplainerContent},
+			{templatepkgaction.ExplainerCustomBase, templatepkgaction.ExplainerCustomContent},
+		}
+
+		err := write(data, templates, "pkg/action/%s/%s")
```
Comment (xh3b4sd) [inline cmd/generate/action/runner.go:95]: The action packages are now also generated.

File: cmd/generate/action/template/cmd/flag.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package template
+package cmd
```
Comment (xh3b4sd) [inline cmd/generate/action/template/cmd/flag.go:1]: The templates got moved due to the new structure we needed for the `pkg/action` templates.

File: cmd/generate/action/runner.go [hunk 4]
```diff
@@ -98,9 +127,38 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
...
+			} else {
+				exists = true
+			}
+
+			var regenerate bool
+			if key.HasGeneratedPrefix(base) {
+				regenerate = true
+			}
+
+			// This means that r.g. custom business logic in explainer.go is
```
Comment (njuettner) [inline cmd/generate/action/runner.go:147]: e.g

Comment (xh3b4sd) [inline cmd/generate/action/runner.go:147]:
```suggestion
			// This means that e.g. custom business logic in explainer.go is
```

**Pull request merged**